api/src/pcapi/core/subscription/api.py

```diff
@@ -103,11 +103,30 @@ def activate_beneficiary(user: users_models.User) -> users_models.User:
     return activate_beneficiary_for_eligibility(user, fraud_check.get_detailed_source(), eligibility)  # type: ignore [arg-type]
 
 
-def has_completed_profile(user: users_models.User) -> bool:
+def _has_completed_profile(user: users_models.User) -> bool:
     mandatory_fields = [user.city, user.activity, user.firstName, user.lastName]
     return all(elem is not None for elem in mandatory_fields)
 
 
+def should_complete_profile(user: users_models.User) -> bool:
+    if _has_completed_profile(user):
+        return False
+    # if the user has a pending DMS subscription, we consider the profile as potentially completed
+    # (it will be completed when the subscription is activated)
+    user_pending_dms_fraud_checks = [
+        fraud_check
+        for fraud_check in user.beneficiaryFraudChecks
+        if fraud_check.type == fraud_models.FraudCheckType.DMS
+        and fraud_check.status in (fraud_models.FraudCheckStatus.PENDING, fraud_models.FraudCheckStatus.STARTED)
+        and fraud_check.resultContent
+        and fraud_check.resultContent["city"] is not None  # TODO: remove when all DMS applications ask for city
+    ]
+    if user_pending_dms_fraud_checks:
+        return False
```

Comment:
une méthode bien nommée pourrait éviter d'écrire le commentaire (ou on peut en profiter pour donner du why) 
```
def _has_completed_profile_in_dms_form() 
   ...

# don't let users with pending dms complete profile on the app, it would make them do another action without making them beneficiary
if _has_completed_profile_in_dms_form():
  return False
return True
`